communitymanager/src/main/java/org/pesmypetcare/communitymanager/managers/ChatManager.java

```diff
@@ -0,0 +1,191 @@
+    private ChatException exception;
+
+    /**
+     * Default constructor.
+     */
+    public ChatManager() {
+        db = FirebaseFirestore.getInstance();
+        exception = null;
+        listener = null;
+    }
+
+    /**
+     * Creates a ChatManager with the given instance of Firestore.
+     * @param firestore The Firestore instance
+     */
+    public ChatManager(FirebaseFirestore firestore) {
+        db = firestore;
+        exception = null;
+        listener = null;
+    }
+
+    /**
+     * Creates a listener for the forum messages.
+     *
+     * @param group The name of the group where the forum belongs
+     * @param forum The name of the group
+     * @param mutableData The mutable message data
+     * @throws ChatException When the chat creation fails
+     */
+    public void createMessageListener(String group, String forum, MutableLiveData<MessageDisplay> mutableData)
+            throws ChatException {
+        if (BuildConfig.DEBUG) {
+            Log.d(TAG, "Path to group name: " + GROUP_NAMES_PATH + group);
+        }
+        System.out.println("HERE 1");
+        db.document(GROUP_NAMES_PATH + group).get().addOnCompleteListener(task -> {
+            System.out.println("HERE 2");
+            if (task.isSuccessful()) {
+                DocumentSnapshot groupDoc = task.getResult();
+                if (groupDoc != null) {
+                    if (groupDoc.exists()) {
+                        groupId = groupDoc.getString("group");
+                        if (BuildConfig.DEBUG) {
+                            Log.d(TAG, "Group ID: " + groupId);
+                        }
+                        try {
+                            System.out.println("HERE");
+                            getForumIdAndCreateListener(group, forum, mutableData);
+                        } catch (ChatException e) {
+                            e.printStackTrace();
+                            exception = e;
+                        }
+                    } else {
+                        exception = new ChatException("No such group");
+                    }
+                } else {
+                    exception = new ChatException("No such group");
+                }
+            } else {
+                exception = new ChatException("Get group failed", task.getException());
+            }
+        });
+        if (exception != null) {
+            if (BuildConfig.DEBUG) {
+                Log.e(TAG, exception.getMessage(), exception);
+            }
+            throw exception;
+        }
+    }
+
+    /**
+     * Gets the forum ID and creates the listener for its messages.
+     * @param group The name of the group where the forum belongs
+     * @param forum The name of the forum
+     * @param mutableData The mutable message data
+     * @throws ChatException When the creation of the listener fails
+     */
+    private void getForumIdAndCreateListener(String group, String forum, MutableLiveData<MessageDisplay> mutableData)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [NCSS for this method is 25 (max allowed is 15).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5538337)